build.gradle

```diff
@@ -73,7 +73,13 @@ ext {
 
 allprojects {
     group 'org.apache.ignite'
-    version = "3.0.0-SNAPSHOT"
+    def gitLastCommitHash = new ByteArrayOutputStream()
+    exec {
+        commandLine "git", "rev-parse", "--short", "HEAD"
+        workingDir = projectDir
+        standardOutput = gitLastCommitHash
+    }
+    version = "3.0.0-SNAPSHOT-$gitLastCommitHash"
```

Comment:
I think we should not change the product version. Rather store the commit hash into a separate place (like file) and let the node read it from this place.